code/modules/power/geothermal/_geothermal.dm

```diff
@@ -1,63 +1,192 @@
 var/global/const/GEOTHERMAL_EFFICIENCY_MOD =             0.2
 var/global/const/GEOTHERMAL_PRESSURE_LOSS =              0.3
-var/global/const/GEOTHERMAL_PRESSURE_CONSUMED_PER_TICK = 0.5
-var/global/const/GEOTHERMAL_PRESSURE_TO_POWER =        100
-var/global/const/MAX_GEOTHERMAL_PRESSURE =            2000
+var/global/const/GEOTHERMAL_PRESSURE_CONSUMED_PER_TICK = 0.05
+var/global/const/GEOTHERMAL_PRESSURE_TO_POWER =          800
+var/global/const/MAX_GEOTHERMAL_PRESSURE =               12000
 
+//
+/particles/geyser_steam
+	icon_state = "smallsmoke"
+	icon       = 'icons/effects/effects.dmi'
+	width      = WORLD_ICON_SIZE * 2
+	height     = WORLD_ICON_SIZE * 6 //Needs to be really tall, otherwise particles stop being drawn outside of the canvas.
+	count      = 64
+	spawning   = 5
+	lifespan   = generator("num", 1 SECOND, 2.5 SECONDS, LINEAR_RAND)
+	fade       = 3 SECONDS
+	fadein     = 0.25 SECONDS
+	grow       = 0.1
+	velocity   = generator("vector", list(0, 0), list(0, 0.2))
+	position   = generator("circle", -6, 6, NORMAL_RAND)
+	//drift      = generator("vector", list(0, -0.2), list(0, 0.2))
+	gravity    = list(0, 0.40)
+	scale      = generator("vector", list(0.3, 0.3), list(1,1), NORMAL_RAND)
+	rotation   = generator("num", -45, 45)
+	spin       = generator("num", -20, 20)
+
+//////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////
 /obj/effect/geyser
-	name = "geothermal vent"
-	desc = "A crack in the ocean floor that occasionally vents gouts of superheated water and steam."
-	icon = 'icons/effects/geyser.dmi'
+	name       = "geothermal vent"
+	desc       = "A vent leading to an underground geothermally heated reservoir, which periodically spew superheated liquid."
+	icon       = 'icons/effects/geyser.dmi'
 	icon_state = "geyser"
-	anchored = TRUE
-	layer = TURF_LAYER + 0.01
+	anchored   = TRUE
+	layer      = TURF_LAYER + 0.01
+	level      = 1 //Goes under floor/plating
+	var/particles/geyser_steam/steamfx
 
-/obj/effect/geyser/Initialize()
+/obj/effect/geyser/Initialize(ml)
 	. = ..()
 	if(prob(50))
 		var/matrix/M = matrix()
 		M.Scale(-1, 1)
 		transform = M
 	set_extension(src, /datum/extension/geothermal_vent)
+	steamfx = new //Prepare our FX
+
+/obj/effect/geyser/proc/do_spout()
+	particles = steamfx
+	particles.spawning = initial(particles.spawning)
+	spawn(6 SECONDS)
```

Comment:
Why not waitfor FALSE?

Reply:
I actually don't remember why I went with this. 
All I remember is that timers would basically shit themselves and I got mad and used spawn instead.
I guess waitfor could work.